pkg/controller/servicemesh/webhook/controller.go

```diff
@@ -0,0 +1,205 @@
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/maistra/istio-operator/pkg/controller/common"
+	"k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var (
+	log         = logf.Log.WithName("controller_webhook")
+	secretNames = []string{
+		"istio.istio-sidecar-injector-service-account",
+		"istio.istio-galley-service-account",
+	}
+
+	injectorWebhookConfigNamePrefix = "istio-sidecar-injector-"
+	galleyWebhookConfigNamePrefix   = "istio-galley-"
+
+	rootCertKey = "root-cert.pem"
+)
+
+// Add creates a new Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &reconciler{ResourceManager: common.ResourceManager{Client: mgr.GetClient(), Log: log}, scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("webhook-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch secrets, but only where name is in secretNames
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			return contains(secretNames, event.Meta.GetName())
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			return contains(secretNames, event.MetaNew.GetName())
+		},
+		// deletion and generic events don't interest us
+		DeleteFunc: func(event event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(event event.GenericEvent) bool {
+			return false
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch MutatingWebhookConfigurations
+	err = c.Watch(&source.Kind{Type: &v1beta1.MutatingWebhookConfiguration{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(ns handler.MapObject) []reconcile.Request {
+			requests := []reconcile.Request{}
+			if strings.HasPrefix(ns.Meta.GetName(), injectorWebhookConfigNamePrefix) {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      secretNames[0],
```

Comment:
Nitpick: can you introduce `sidecarInjectorServiceAccountSecretName` instead of referring to `secretNames[0]`? It would be a bit nicer to anyone reading this code.